src/lib/mina_base/parties.ml

```diff
@@ -0,0 +1,140 @@
+open Core
+
+[%%versioned
+module Stable = struct
+  module V1 = struct
+    type t =
+      { parties: Party.Signed.Stable.V1.t * Party.Stable.V1.t list
+      ; protocol_state: Snapp_predicate.Protocol_state.Stable.V1.t }
+    [@@deriving sexp, compare, eq, hash, yojson]
+
+    let to_latest = Fn.id
+
+    let version_byte = Base58_check.Version_bytes.snapp_command
+
+    let description = "Parties"
+  end
+end]
+
+include Codable.Make_base58_check (Stable.Latest)
+
+module Valid = struct
+  module Stable = Stable
+
+  type t = Stable.Latest.t
+end
+
+let check (t : t) =
+  let p, _ = t.parties in
+  Token_id.(equal default) p.data.body.token_id
+  && Sgn.(equal Neg) p.data.body.delta.sgn
+
+let parties (t : t) : Party.t list =
+  let p, ps = t.parties in
+  { authorization= Control.Signature p.authorization
+  ; data= {p.data with predicate= Party.Predicate.Nonce p.data.predicate} }
+  :: ps
+
+let fee_lower_bound (t : t) : Currency.Fee.t =
```

Comment:
I'm not convinced that this name is accurate: won't the 'fee-payer' end up giving all of its delta to the block producer if the other updates fail, whereas some of its delta may be taken by other parties if they succeed?

Reply:
Can you call this `fee_lower_bound_exn`? A doc-comment saying that this may `raise AssertionError` if `check t <> true` would also be good.